src/IECoreMaya/ToMayaMeshConverter.cpp

```diff
@@ -344,6 +346,123 @@ bool ToMayaMeshConverter::doConversion( IECore::ConstObjectPtr from, MObject &to
 		}
 	}
 
+	/// Add corners and edge creases
+
+	MUintArray cornerIdsMaya;
+	MDoubleArray cornerSharpnessMaya;
+
+	const IECore::IntVectorData *cornerIdsData = mesh->cornerIds();
+	const std::vector<int> cornerIds = cornerIdsData->readable();
+
+	if( !cornerIds.empty() )
+	{
+		const IECore::FloatVectorData *cornerSharpnessesData = mesh->cornerSharpnesses();
+		const std::vector<float> cornerSharpnesses = cornerSharpnessesData->readable();
+
+		for( size_t i = 0; i < cornerIds.size(); ++i )
+		{
+			cornerIdsMaya.append( cornerIds[i] );
+			cornerSharpnessMaya.append( cornerSharpnesses[i] );
+		}
+
+		s = fnMesh.setCreaseVertices( cornerIdsMaya, cornerSharpnessMaya );
+		if( !s )
+		{
+			IECore::msg( IECore::Msg::Warning, "ToMayaMeshConverter::doConversion", boost::format( "Failed to set crease vertices with message: %s" ) % s.errorString().asChar() );
+		}
+	}
+
+	MUintArray edgeIdsMaya;
+	MDoubleArray creaseSharpnessMaya;
+
+	const IECore::IntVectorData *creaseIdsData = mesh->creaseIds();
+	const std::vector<int> creaseIds = creaseIdsData->readable();
+
+	if( !creaseIds.empty() )
+	{
+		const IECore::FloatVectorData *creaseSharpnessesData = mesh->creaseSharpnesses();
+		const std::vector<float> creaseSharpnesses = creaseSharpnessesData->readable();
+
+		const IECore::IntVectorData *creaseLengthsData = mesh->creaseLengths();
+		const std::vector<int> creaseLengths = creaseLengthsData->readable();
+
+		// Cortex stores vertex ids to specify creases. Maya uses edge ids
+		// instead. The following handles the conversion.
+
+		MDagPath dag;
+		fnMesh.getPath( dag );
+		MItMeshVertex vertexIt( dag );
+
+		int offset = 0;
+		for( size_t i = 0; i < creaseLengths.size(); ++i )
+		{
+			int length = creaseLengths[i];
+
+			for( int j = 1; j < length; ++j )
+			{
+				int vertexId1 = creaseIds[offset + j - 1];
+				int vertexId2 = creaseIds[offset + j];
+
+				// Find the edgeId for the edge connecting the two vertices.
+				// Currenty, this is done by getting the set of connected edges
+				// for each vert and then intersecting the two sets - assuming
+				// that exactly one edge will be present in BOTH lists.
+				// \todo: Maybe there's a better way to do this?
```

Comment:
It seems that USD is just querying `getConnectedEdges()` for the first vertex, and then iterating those edges until `getOppositeVertex()` gives the index of the second vertex. If this will work for us too, it seems preferable  because it's linear rather than quadratic in the number of connected edges. https://github.com/PixarAnimationStudios/USD/blob/master/third_party/maya/lib/usdMaya/translatorMesh_SubDiv.cpp#L224-L243.

Reply:
What I've done in 4160748 should be ok?